### PR TITLE
Adds const to Typedef.toString

### PR DESCRIPTION
**Pull request by denizzzka**

### CODE REVIEW

**Comment by jmdavis** [comment]
What happens if the type being wrapped does not have a `toString` which is `const`? Glancing at the code, I expect that it won't compile. So, at minimum, that case must be tested for, but it's probably the case that it's not possible to make `Typedef.toString` `const` without getting into the whole mess of duplicating it and using static ifs to put `const` on it or not based on whether it can work as `const`, which gets ugly fast.

**Comment by Bolpat** [comment]
Use `tostring(this Me)()`. Then `Me` becomes the type of the object including qualifiers. That's the temple version of `inout`. For the language reference, see [Template This Parameter](https://dlang.org/spec/template.html#template_this_parameter).

**Comment by n8sh** [comment]
This issue isn't uncommon. It seems that the best thing might be for `const` to be inferred like `@nogc` and `nothrow`, although I am sure this has been discussed before.

**Comment by Bolpat** [comment]
@n8sh In generic code, inferring things is always the way to go. There is still a difference between attributes and `const`: The latter is a type qualifier. I'm not sure if inferring it for templates *by default* is good. The necessity is low as it can be done on demand.

File: std/typecons.d [hunk 1]
```diff
@@ -7177,7 +7177,7 @@ struct Typedef(T, T init = T.init, string cookie=null)
     /**
      * Convert wrapped value to a human readable string
      */
-    string toString()
+    string toString() const
```
Comment (Bolpat) [inline std/typecons.d]: Use `string toString(this T)()`

File: std/typecons.d [hunk 2]
```diff
@@ -7187,7 +7187,7 @@ struct Typedef(T, T init = T.init, string cookie=null)
     }
 
     /// ditto
-    void toString(W)(ref W writer, const ref FormatSpec!char fmt)
+    void toString(W)(ref W writer, const ref FormatSpec!char fmt) const
```
Comment (Bolpat) [inline std/typecons.d]: Use `void toString(this T, W)(ref W writer, const ref FormatSpec!char fmt)`

**Comment by denizzzka** [comment]
@Bolpat thanks!

**Comment by MetaLang** [comment]
Can confirm what @jmdavis said; it just won't work. I also have not thought to try `toString(this T)()` though, which probably _will_ work. Good idea @Bolpat.

**Comment by denizzzka** [comment]
This PR can be merged?

**Pull request merged**